lib/user.sh

```diff
@@ -167,4 +168,11 @@ user.current-shell-init-file() {
   echo "${init_file}"
 }
 
+## :TBD: As one I recently had need of, where's `user.home-dir`?
+## 
+## user.home-dir() {
+##   local user="${1:-$( whoami )}"
+##   printf '%s\n' $(bash -c "cd ~$(printf %q "${user}") && pwd")
+## }
+
```

Comment:
This is quite clever :) 

```suggestion
  user.home-dir() {
    local user
    user="${1:-$( whoami )}";    
    printf '%s\n' $(bash -c "cd ~$(printf %q "${user}") && pwd")
  }
```

Reply:
Thought it was clever too. Forgot what SE solution I "borrowed" it from though so no attribution for them :(